src/data/seq/seq.lean

```diff
@@ -42,9 +42,18 @@ instance : inhabited (seq α) := ⟨nil⟩
 def cons (a : α) : seq α → seq α
 | ⟨f, al⟩ := ⟨some a :: f, λn h, by {cases n with n, contradiction, exact al h}⟩
 
+@[simp] lemma val_cons (s : seq α) (x : α) : (cons x s).val = some x :: s.val :=
+by cases s; refl
+
 /-- Get the nth element of a sequence (if it exists) -/
 def nth : seq α → ℕ → option α := subtype.val
 
+@[simp] lemma nth_cons_zero (s : seq α) (x : α) : nth (cons x s) 0 = some x :=
```

Comment:
I'm already adding these in #15830. You might want to make this PR depend on that one, see below.